aptos-move/framework/aptos-framework/sources/aggregator/optional_aggregator.move

```diff
@@ -9,26 +9,27 @@ module aptos_framework::optional_aggregator {
 
     friend aptos_framework::coin;
 
-    // These error codes are produced by `Aggregator` and used by `Integer` for
-    // consistency.
+    /// When the value of aggregator underflows (goes below zero). Raised by native code.
     const EAGGREGATOR_OVERFLOW: u64 = 1;
+
+    /// When aggregator feature is not supported. Raised by native code.
     const EAGGREGATOR_UNDERFLOW: u64 = 2;
 
-    /// Wrapper around integer to have a custom overflow limit. Note that
-    /// Move has no traits (and trait bounds), so integer value must be u128.
-    /// `Integer` provides API to add/subtract and read, just like `Aggregator`.
+    /// Wrapper around integer with a custom overflow limit. Supports add, subtract and read just like `Aggregator`.
     struct Integer has store {
         value: u128,
         limit: u128,
     }
 
+    /// Creates a new integer
```

Comment:
This is rather a lazy comment. Can you explain a bit more on what this struct is for?